include/hybridmap/HybridSynthesisMapper.hpp

```diff
@@ -44,103 +45,186 @@ class HybridSynthesisMapper : public NeutralAtomMapper {
-   * @param arch Neutral atom architecture.
-   * @param params Optional mapper configuration parameters.
+   * @brief Create a HybridSynthesisMapper configured for a neutral-atom device.
+   *
+   * @param arch Neutral-atom device architecture used for mapping and
+   * scheduling.
+   * @param params Optional mapper configuration parameters; defaults to
+   * MapperParameters().
+   * @param bufferSize Number of operations to hold in the internal buffer
+   * before flushing to the synthesized circuit (0 disables buffering).
    */
   explicit HybridSynthesisMapper(
       const NeutralAtomArchitecture& arch,
-      const MapperParameters& params = MapperParameters())
-      : NeutralAtomMapper(arch, params) {}
+      const MapperParameters& params = MapperParameters(),
+      const uint32_t bufferSize = 0)
+      : NeutralAtomMapper(arch, params), bufferSize(bufferSize) {}
 
   // Functions
 
   /**
-   * @brief Initialize synthesized and mapped circuits and mapping structures.
-   * @param nQubits Number of logical qubits to synthesize.
+   * @brief Initialize internal circuits and mapping state for a given number of
+   * logical qubits.
+   *
+   * Sets up the synthesized and buffered quantum computations, the mapped
+   * circuit storage, the logical-to-physical mapping, and records the original
+   * mapping; marks the mapper as initialized.
+   *
+   * @param nQubits Number of logical qubits to prepare.
+   * @throws std::runtime_error if `nQubits` exceeds the architecture's
+   * available positions.
    */
   void initMapping(const size_t nQubits) {
     if (nQubits > arch->getNpositions()) {
       throw std::runtime_error("Not enough qubits in architecture.");
     }
     mappedQc = qc::QuantumComputation(arch->getNpositions());
     synthesizedQc = qc::QuantumComputation(nQubits);
+    bufferedQc = qc::QuantumComputation(nQubits);
     mapping = Mapping(nQubits);
     initialized = true;
+    originalMapping = mapping;
   }
 
   /**
-   * @brief Complete a (re-)mapping of the synthesized circuit to hardware.
-   * @param initMapping Initial mapping heuristic (defaults to Identity).
+   * @brief Remap the synthesized circuit onto the stored original hardware
+   * mapping.
+   *
+   * If `includeBuffer` is true, the remap uses the combined
+   * synthesized-plus-buffered circuit returned by getSynthesizedQc(); otherwise
+   * it remaps only the internal synthesizedQc (excluding buffered operations).
+   * The result is mapped into originalMapping.
+   *
+   * @param includeBuffer When true, include buffered operations in the remap.
    */
-  void completeRemap(const InitialMapping initMapping = Identity) {
-    auto qcCopy = synthesizedQc;
-    map(qcCopy, initMapping);
+  void completeRemap(const bool includeBuffer = true) {
+    if (includeBuffer) {
+      auto copyQC = getSynthesizedQc();
+      map(copyQC, originalMapping);
+    } else {
+      auto temp = synthesizedQc;
+      this->map(temp, originalMapping);
```

Comment:
```suggestion
      map(temp, originalMapping);
```
Consistency with the other branch.